### PR TITLE
Blog about Apache Camel and the log4j security issue

### PR DESCRIPTION
**Pull request by davsclaus**

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

**Comment by tadayosi** [comment]
FYI, for Spring Boot we need to wait until Dec 23, 2021 to get the fixed versions `v2.5.8` & `v2.6.2`:
https://spring.io/blog/2021/12/10/log4j2-vulnerability-and-spring-boot

**Comment by davsclaus** [comment]
Okay but that is a spring boot problem, not a Camel problem.

**Comment by davsclaus** [comment]
And on spring boot it depends on which logger you choose to use, they primary use logback as their preferred logger.

**Comment by tadayosi** [comment]
Yes, that's correct.  I thought that we have a mention to Quarkus already so it might be nice to have another mention to SB as well.

**Review by astefanutti — APPROVED** [review]

**Comment by davsclaus** [comment]
> Yes, that's correct. I thought that we have a mention to Quarkus already so it might be nice to have another mention to SB as well.

Ah yeah that is a good idea - I update the blog with details about camel on spring boot

**Comment by davsclaus** [comment]
Camel on Apache Kafka Connectors.

Then I read that Kafka is using the old log4j v1 - but would like to know more what is the status on Kafka.
However again its a Kafka problem and not CKC but would be good to refer to their announcement or something.
For Quarkus then I have seen a tweet, not a blog post on their website.

**Review by aldettinger — APPROVED** [review]

**Review by tadayosi — APPROVED** [review]

**Comment by bvn13** [comment]
I found only one component depending log4j-core not in test. It is camel-jbang-core. Have you investigate it for vulnerability?

**Comment by davsclaus** [comment]
Ah yeah CamelJBang is only a developer tool, not for production usage, and it has only been added in non LTS releases, but its coming in 3.14.0.

**Review by squakez — APPROVED** [review]

**Review by bvn13 — APPROVED** [review]

**Review by orpiske — APPROVED** [review]

**Comment by orpiske** [comment]
> Camel on Apache Kafka Connectors.
> 
> Then I read that Kafka is using the old log4j v1 - but would like to know more what is the status on Kafka. However again its a Kafka problem and not CKC but would be good to refer to their announcement or something. For Quarkus then I have seen a tweet, not a blog post on their website. https://twitter.com/QuarkusIO/status/1469279468829265922?s=20

CKC uses log4j2 for testing. I'll search for additional details for Kafka and I'll share here.

**Comment by orpiske** [comment]
> > Camel on Apache Kafka Connectors.
> > Then I read that Kafka is using the old log4j v1 - but would like to know more what is the status on Kafka. However again its a Kafka problem and not CKC but would be good to refer to their announcement or something. For Quarkus then I have seen a tweet, not a blog post on their website. https://twitter.com/QuarkusIO/status/1469279468829265922?s=20
> 
> CKC uses log4j2 for testing. I'll search for additional details for Kafka and I'll share here.

There is no [official statement from Kafka](https://kafka.apache.org/cve-list) at this moment. Although they are working on migrating to log4j2 via [KAFKA-9366](https://issues.apache.org/jira/browse/KAFKA-9366), this has not been merged yet (details on the ticket). Log4j v1 [does not seem to be impacted by this CVE](https://twitter.com/ceki/status/1469449618316533762).

**Review by gzurowski — APPROVED** [review]
LGTM 👍

**Comment by davsclaus** [comment]
I update the blog with details about CKC and a section about two camel components that transitive has dependency on log4j-core (not only for testing).

File: content/blog/2021/12/log4j2/index.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+In the upcoming LTS releases 3.14.0, 3.11.5, and 3.7.7 we have upgraded to
+log4j 2.15.0. For future releases then we plan to filter out testing
+dependencies in the `camel-dependencies` BOM, meaning that `log4j2-version`
+will no longer be included.
+
+### 3rd party components with transitive dependency on log4j-core
+
+There are two Camel components, camel-nsq and camel-corda which has
+dependency on log4j-core by their 3rd party library. Those libraries
+does not active use log4j-core, but rely only on log4j-api, so the maintainers
```
Comment (orpiske) [inline content/blog/2021/12/log4j2/index.md:38]: Looks like a typo: shouldn't it be "Those libraries do not actively use"?

Comment (davsclaus) [reply content/blog/2021/12/log4j2/index.md:38]: Ah yeah good to catch those typos, feel free to edit as this is a blog announcement from all of us.

Comment (orpiske) [reply content/blog/2021/12/log4j2/index.md]: Thanks. I didn't know I could edit them directly. I wrote my suggested fixes via commit.

**Comment by davsclaus** [comment]
Added a note about Camel 2

**Review by zhfeng — APPROVED** [review]

**Review by zbendhiba — APPROVED** [review]
Thanks @davsclaus

**Comment by davsclaus** [comment]
Added a note about Apache Karaf because its affected

**Comment by bvn13** [comment]
> Added a note about Apache Karaf because its affected

Will new release of Camel 3.14.0 wait until Karaf be fixed by its maintainers?

**Comment by davsclaus** [comment]
No they are independent projects. We only uses karaf for testing camel-karaf, not as a required runtime version, or that we ship apache karaf as part of a camel release

**Comment by davsclaus** [comment]
Added note about camel jbang, and also link to karaf about their work on next release with fix

**Pull request merged**